### ISSUE TITLE
Can not resume downloading blockchain data

### ISSUE THREAD

**Issue by UniBanker** [btcsuite/btcd#1119]
In https://github.com/btcsuite/btcd/commit/31444f589086ba7979b1a023ed412e44a2c70635 I believe this behavior is broken, causing the genesis block to actually be the 2nd block, as stored in the database. As a result a fresh sync of the chain and following resume is failing.

**Comment by jsmootiv** [comment]
Im seeing this issue also. When I start the node for the first time it syncs fine.

But if I shut down and restart I get:

`assertion failed: initChainState: Expected first entry in block index to be genesis block, found 00000000b873e79784647a6c82962c70d228557d24a747ea4d1b8bbe878e1206`

**Comment by vadimicus** [comment]
Have the same issue on testnet and mainnet

Tried on:

* ubuntu 16.06
```
go get -u github.com/Masterminds/glide && \
    git clone https://github.com/btcsuite/btcd $GOPATH/src/github.com/btcsuite/btcd && \
    cd $GOPATH/src/github.com/btcsuite/btcd && \
    glide install && \
    go install . ./cmd/..
```

* macOS High Sierra 10.13.2

* I also buit [docker](multyio/btcd:latest)

All builds after correct termination throwing the same error:

```
ERR] BTCD: Unable to start server on [:18333]: assertion failed: initChainState: Expected first entry in block index to be genesis block, found 00000000b873e79784647a6c82962c70d228557d24a747ea4d1b8bbe878e1206
```

**Comment by vadimicus** [comment]
The same issue is on the **MainNet**
```
btcd-testnet_1  | 2018-02-06 11:40:13.439 [INF] BTCD: Version 0.12.0-beta
btcd-testnet_1  | 2018-02-06 11:40:13.442 [INF] BTCD: Loading block database from '/root/.btcd/mainnet/blocks_ffldb'
btcd-testnet_1  | 2018-02-06 11:40:13.475 [INF] BTCD: Block database loaded
btcd-testnet_1  | 2018-02-06 11:40:13.479 [INF] INDX: Transaction index is enabled
btcd-testnet_1  | 2018-02-06 11:40:13.484 [INF] CHAN: Loading block index...
btcd-testnet_1  | 2018-02-06 11:40:13.591 [ERR] BTCD: Unable to start server on [:8333]: assertion failed: initChainState: Expected first entry in block index to be genesis block, found 00000000839a8e6886ab5951d76f411475428afc90947ee320161bbf18eb6048
btcd-testnet_1  | 2018-02-06 11:40:13.591 [INF] BTCD: Gracefully shutting down the database...
btcd-testnet_1  | 2018-02-06 11:40:13.711 [INF] BTCD: Shutdown complete
```

**Comment by vadimicus** [comment]
Hot fix is:
```
go get -u github.com/Masterminds/glide && \
    git clone https://github.com/btcsuite/btcd $GOPATH/src/github.com/btcsuite/btcd && \
    cd $GOPATH/src/github.com/btcsuite/btcd && \
    git checkout a4d22d8384806ca60e537108ed1672d330e1bda2 . && \
    git commit -m "sync working =)" && \
    glide install && \
    go install . ./cmd/...
```

**Comment by shamatar** [comment]
On 9th of February after pulling 

HEAD is now at a4d22d8 blockchain: Delete block headers from old block index bucket.
commit a4d22d8384806ca60e537108ed1672d330e1bda2
Author: Jim Posen <jimpo@coinbase.com>
Date:   Tue Sep 5 19:22:16 2017 -0700

    blockchain: Delete block headers from old block index bucket.

    As part of the migration to the new block index bucket, also delete
    the redundant data from the old bucket.

commit ad00e7ff82020979ba23252085b2a2f09608f335

same error

**Comment by jrick** [comment]
Fixed by #1122.
